libs/model/src/models/token.ts

```diff
@@ -20,16 +20,29 @@ export default (
   sequelize.define<TokenInstance>(
     'Token',
     {
-      name: { type: Sequelize.STRING, primaryKey: true },
-      icon_url: { type: Sequelize.STRING, allowNull: true },
-      description: { type: Sequelize.STRING, allowNull: true },
+      // derivable when event received
+      token_address: { type: Sequelize.STRING, primaryKey: true },
+      namespace: {
+        type: Sequelize.STRING,
+        references: {
```

Comment:
Foreign keys/references are not defined in the model.

Reply:
We should actually keep this. There is no way currently to specify a non-id foreign key in the associations file

Reply:
I requested that we do it this way because it means we can keep the chain event-derived models as disconnected as possible from our internal DB keys/models. The goal is to eventually use something like SubQuery which would mean we would have a separate DB for chain-event models. Therefore we would need to query this separate DB using `namespace` rather than the community_id or some surrogate key (which won't exist in the SubQuery DB).

Having a fkey to `namespace` is a half-step in that direction and makes it easier to migrate the models in the future.

Reply:
this is a common scenario in micro-service architectures where each service has a private db. if we are trying to decouple communities from tokens in different dbs (logically at the moment), let's keep both tables disconnected then, no need for artificial FKs.... we can always find the entity by PK or namespace attribute by querying the other "service"

Reply:
I'm fine with that.